bindgen/codegen.ml

```diff
@@ -22,11 +24,35 @@ let write_dune_file (dune : Dunefile.t) =
   Format.fprintf fmt "\n%!"
 
 let write_caml_files caml (dune : Dunefile.t) =
+  let runtime_ocaml = read_file "../bindgen/runtime.ml" in
   with_file dune.caml_file_name @@ fun fmt ->
   Format.fprintf fmt "(* automatically generated by ocaml-bindgen 0.0.1 *)\n";
+  Format.fprintf fmt "%s\n" runtime_ocaml;
   Format.fprintf fmt "%s\n%!" (Format.asprintf "%a" Pprintast.structure caml)
 
 let write_c_files (c : C.program) (dune : Dunefile.t) =
   with_file dune.c_file_name @@ fun fmt ->
   Format.fprintf fmt "/* automatically generated by ocaml-bindgen 0.0.1 */\n";
-  Format.fprintf fmt "%s\n%!" (Format.asprintf "%a" C.pp c)
+  Format.fprintf fmt "\n%s\n%!" (Format.asprintf "%a" C.pp c);
+  Format.fprintf fmt
+    {|#include <stdlib.h>
+value bindgen_alloc(value caml_size) {
+    CAMLparam1(caml_size);
+
+    // Convert OCaml integer to C size
+    size_t size = Int_val(caml_size);
+    printf("Allocated size %%ld \n", size);
+
+    void* ptr = malloc(sizeof(size));
+    if (ptr == NULL) {
+        // TODO: handle allocation failure
+        CAMLreturn(Val_unit);
+    }
+    
+    // Wrap the pointer as an OCaml value
+    CAMLreturn(caml_copy_nativeint(ptr));
+}
+
+void bindgen_free(value caml_addr) {
+    free(Nativeint_val(caml_addr));
+} |}
```

Comment:
We end up needing this to handle the creation of pointers. This is effectively replacing what we do inside the `of_value` functions. In my example its just for an `int` but it could also be for `struct`s.

I can see it being a bit slower, so there might be a way to specialise certain cases to keep the current behaviour, wherein you pass a record by value (`doggo -> unit`) and it stack allocates it on the C side a.la https://github.com/ocaml-sys/ocaml-bindgen/issues/1#issuecomment-1936865219